code/modules/clothing/clothing.dm

```diff
@@ -421,30 +415,41 @@ BLIND     // can't see anything
 //Proc that opens and closes jackets.
 /obj/item/clothing/suit/proc/adjustsuit(var/mob/user)
 	if(!ignore_suitadjust)
-		if(!user.canmove || user.stat || user.restrained())
-			return
-		if(suit_adjusted)
-			var/flavour = "close"
-			icon_state = initial(icon_state)
-			item_state = initial(item_state)
-			if(adjust_flavour)
-				flavour = "[copytext(adjust_flavour, 3, lentext(adjust_flavour) + 1)] up" //Trims off the 'un' at the beginning of the word. unzip -> zip, unbutton->button.
-			user << "You [flavour] \the [src]."
-			suit_adjusted = 0 //Suit is no longer adjusted.
-		else
-			var/flavour = "close"
-			icon_state += "_open"
-			item_state += "_open"
-			if(adjust_flavour)
-				flavour = "[adjust_flavour]"
-			user << "You [flavour] \the [src]."
-			suit_adjusted = 1 //Suit's adjusted.
-
-		usr.update_inv_wear_suit()
+		if(!user.incapacitated())
+			if(!(HULK in user.mutations))
+				if(suit_adjusted)
+					var/flavour = "close"
+					icon_state = copytext(icon_state, 1, findtext(icon_state, "_open")) /*Trims the '_open' off the end of the icon state, thus avoiding a case where jackets that start open will
+																							end up with a suffix of _open_open if adjusted twice, since their initial state is _open. */
+					item_state = copytext(item_state, 1, findtext(item_state, "_open"))
+					if(adjust_flavour)
+						flavour = "[copytext(adjust_flavour, 3, lentext(adjust_flavour) + 1)] up" //Trims off the 'un' at the beginning of the word. unzip -> zip, unbutton->button.
+					user << "You [flavour] \the [src]."
+					suit_adjusted = 0 //Suit is no longer adjusted.
+				else
+					var/flavour = "close"
+					icon_state += "_open"
+					item_state += "_open"
+					if(adjust_flavour)
+						flavour = "[adjust_flavour]"
+					user << "You [flavour] \the [src]."
+					suit_adjusted = 1 //Suit's adjusted.
+			else
+				if(contents) //If the suit's got any storage capability...
+					for(var/obj/item/O in contents) //AVOIDING ITEM LOSS. Check through everything that's stored in the jacket and see if one of the items is a pocket.
+						if(istype(O, /obj/item/weapon/storage/internal)) //If it's a pocket...
+							if(O.contents) //Check to see if the pocket's got anything in it.
+								for(var/obj/item/I in O.contents) //Dump the pocket out onto the floor below the user.
+									I.forceMove(user.loc)
+
+				user.visible_message("<span class='warning'>[user] bellows, [pick("shredding", "ripping open", "tearing off")] their jacket in a fit of rage!</span>","<span class='warning'>You accidentally [pick("shred", "rend", "tear apart")] \the [src] with your [pick("excessive", "extreme", "monstrous", "ridiculous", "unreal")] [pick("power", "strength")]!</span>")
+				qdel(src) //Now that the pockets have been emptied, we can safely destroy the jacket.

```

Comment:
This should probably call unEquip first to avoid any oddities with references not being cleared, also this is going to let you remove NODROP equipment.